storage/aptosdb/src/fake_aptosdb.rs

```diff
@@ -0,0 +1,555 @@
+    }
+
+    fn get_block_info_by_height(&self, height: u64) -> Result<(Version, Version, NewBlockEvent)> {
+        self.inner.get_block_info_by_height(height)
+    }
+
+    fn get_last_version_before_timestamp(
+        &self,
+        timestamp: u64,
+        ledger_version: Version,
+    ) -> Result<Version> {
+        self.inner
+            .get_last_version_before_timestamp(timestamp, ledger_version)
+    }
+
+    fn get_latest_epoch_state(&self) -> Result<EpochState> {
+        self.inner.get_latest_epoch_state()
+    }
+
+    fn get_prefixed_state_value_iterator(
+        &self,
+        key_prefix: &StateKeyPrefix,
+        cursor: Option<&StateKey>,
+        version: Version,
+    ) -> Result<Box<dyn Iterator<Item = anyhow::Result<(StateKey, StateValue)>> + '_>> {
+        self.inner
+            .get_prefixed_state_value_iterator(key_prefix, cursor, version)
+    }
+
+    fn get_latest_ledger_info_option(&self) -> Result<Option<LedgerInfoWithSignatures>> {
+        self.inner.get_latest_ledger_info_option()
+    }
+
+    fn get_latest_state_checkpoint_version(&self) -> Result<Option<Version>> {
+        self.inner.get_latest_state_checkpoint_version()
+    }
+
+    fn get_state_snapshot_before(
+        &self,
+        next_version: Version,
+    ) -> Result<Option<(Version, HashValue)>> {
+        self.inner.get_state_snapshot_before(next_version)
+    }
+
+    fn get_account_transaction(
+        &self,
+        address: aptos_types::PeerId,
+        seq_num: u64,
+        include_events: bool,
+        ledger_version: Version,
+    ) -> Result<Option<TransactionWithProof>> {
+        self.inner
+            .get_account_transaction(address, seq_num, include_events, ledger_version)
+    }
+
+    fn get_account_transactions(
+        &self,
+        address: aptos_types::PeerId,
+        seq_num: u64,
+        limit: u64,
+        include_events: bool,
+        ledger_version: Version,
+    ) -> Result<aptos_types::transaction::AccountTransactionsWithProof> {
+        self.inner
+            .get_account_transactions(address, seq_num, limit, include_events, ledger_version)
+    }
+
+    fn get_state_proof_with_ledger_info(
+        &self,
+        known_version: u64,
+        ledger_info: LedgerInfoWithSignatures,
+    ) -> Result<StateProof> {
+        self.inner
+            .get_state_proof_with_ledger_info(known_version, ledger_info)
+    }
+
+    fn get_state_proof(&self, known_version: u64) -> Result<StateProof> {
+        self.inner.get_state_proof(known_version)
+    }
+
+    fn get_state_value_by_version(
+        &self,
+        state_key: &StateKey,
+        version: Version,
+    ) -> Result<Option<StateValue>> {
+        let access_path = AccessPath::try_from(state_key.clone())?;
+        let account_address = access_path.address;
+        let struct_tag = access_path.get_struct_tag();
+
+        if (account_address != aptos_test_root_address() && account_address != AccountAddress::ONE)
```

Comment:
Sure, forge seems to checks the account balance for which it needs the coin resource, so it access the coin state value. We have that information the AptosDB because we wrote the whole writeset from actual genesis execution there, so we check and go to DB. 

Also, added code comments.